templates/e2e-llm-workflows/README.md

```diff
@@ -446,26 +496,27 @@ from src.utils import download_files_from_bucket
 ```python
 # Locations
 artifacts_dir = '/mnt/cluster_storage'  # storage accessible by head and worker nodes
-model = 'meta-llama/Meta-Llama-3-8B-Instruct'
-uuid = 'gokum:ueewk'  # REPLACE with your NAME + MODEL ID (from Job logs)
-artifacts_path = (
-    f"{os.environ['ANYSCALE_ARTIFACT_STORAGE'].split(os.environ['ANYSCALE_CLOUD_STORAGE_BUCKET'])[-1][1:]}"
-    f"/lora_fine_tuning/{model}:{uuid}")
+model_id = model_info.id 
+artifacts_path = f"{os.environ['ANYSCALE_ARTIFACT_STORAGE']}/lora_fine_tuning/{model_id}"
```

Comment:
Actually no! The models API doesn't know about our checkpoint forwarding. So the `storage_uri` in `model_info` looks like :

```
s3://anyscale-production-data-cld-ldm5ez4edlp7yh4yiakp2u294w/org_4snvy99zwbmh4gbtk64jfqggmj/cld_ldm5ez4edlp7yh4yiakp2u294w/artifact_storage/eric__tang/llmforge-finetuning/meta-llama/Meta-Llama-3-8B-Instruct/TorchTrainer_2024-09-10_18-12-20/epoch-0
```

This happens because we treat checkpoint forwarding and the model registry insert as two independent callbacks so they don't know about each other.  Will need to be fixed on the llm-forge side, but this is not bad for now I think?

Reply:
hmm this is definitely something we should fix. My assumption was the checkpoint upload to s3 is just part of the model registry insert (which is what you'd expect if you were using the sdk from outside llmforge)

Reply:
how do you envision the model registry works if you don't use llmforge

Reply:
> My assumption was the checkpoint upload to s3 is just part of the model registry insert  (which is what you'd expect if you were using the sdk from outside llmforge)

Hmmm good point. Yeah right now these are distinct operations that do not know about each other. The lora checkpoint forwarding has been treated as a rayllm detail separate from the model registry insert. But this can indeed be done at model insert time. 

> (which is what you'd expect if you were using the sdk from outside llmforge)

I agree that when they interact with the SDK, they'd want the model metadata to be self contained for use in inference. This is not the case right now. 

> how do you envision the model registry works if you don't use llmforge

Hmm thinking about this, I'm wondering if the forwarding can indeed be done in the product backend before the model insertion in the DB? That way this API is self contained in registering a model for use with rayllm. Otherwise llmforge will need to encode this information and you end up with corner cases like this.

Reply:
Tracking here: https://anyscale1.atlassian.net/browse/AE-1128 

For the current template, I'm gonna retain the existing code.